src/command/publish.rs

```diff
@@ -1,15 +1,47 @@
+use self::Access::{Public, Restricted};
 use command::utils::{find_pkg_directory, set_crate_path};
 use error::Error;
 use npm;
 use slog::Logger;
 use std::path::PathBuf;
 use std::result;
+use std::str::FromStr;
 use PBAR;
 
+#[derive(Debug)]
+pub enum Access {
+    Public,
+    Restricted,
+}
+
+impl FromStr for Access {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Error> {
+        return match s {
+      "public" => Ok(Access::Public),
+      "restricted" => Ok(Access::Restricted),
+      "private" => Ok(Access::Restricted),
```

Comment:
doing something a little unconventional here but i seriously cannot believe that it's "restricted" and not "private" so i kiiiiiinda want to have this in here, but if ya'll object i'm cool with removing it